src/impl/select.js

```diff
@@ -45,7 +44,7 @@ export function doAlts( // eslint-disable-line
     }
   }
 
-  if (!result && has(options, 'default') && flag.value) {
+  if (!result && flag.value && has(options, 'default')) {
```

Comment:
if the options.default doesn't exist we don't need to check for flag.value, this is intended behavior